code/parachain/frame/crowdloan-rewards/src/lib.rs

```diff
@@ -343,34 +343,54 @@ pub mod pallet {
 			Ok(Pays::No.into())
 		}
 
+		/// Populates the `Rewards` while updating `TotalRewards` and `TotalContributors`
+		///
+		/// If a reward already exits, the reward and respective totals will be updated to account
+		/// for the new values.
+		///
+		/// # Errors
+		/// * `AlreadyInitialized` - The crowdloan has been set to initialize, population may no
+		///   longer commence
+		/// * `ArithmeticError` - Overflow/Underflow detected while calculating totals
 		pub(crate) fn do_populate(
 			rewards: Vec<(RemoteAccountOf<T>, RewardAmountOf<T>, VestingPeriodOf<T>)>,
 		) -> DispatchResult {
 			ensure!(!VestingTimeStart::<T>::exists(), Error::<T>::AlreadyInitialized);
-			rewards
-				.into_iter()
-				.for_each(|(remote_account, account_reward, vesting_period)| {
-					// This will eliminate duplicated entries.
-					Rewards::<T>::insert(
-						remote_account,
-						Reward {
-							total: account_reward,
-							claimed: T::Balance::zero(),
-							vesting_period,
+
+			let mut total_rewards: T::Balance = TotalRewards::<T>::get();
+			let mut total_contributors: u32 = TotalContributors::<T>::get();
+
+			for (remote_account, account_total, vesting_period) in rewards.into_iter() {
+				Rewards::<T>::try_mutate_exists::<_, _, DispatchError, _>(
+					remote_account,
+					|reward| match reward {
+						Some(reward) => {
+							total_rewards = total_rewards.checked_sub(&reward.total).expect(
+								"TotalRewards is greater than or equal to reward.total; QED",
```

Comment:
is there an assertion somewhere for this? or is this just an invariant i'm not aware of

Reply:
f9d8a8b6556ba7731ef70de608f3b9d2a5428d05